pkgs/development/compilers/reason/default.nix

```diff
@@ -13,11 +13,27 @@ stdenv.mkDerivation rec {
     sha256 = "0m6ldrci1a4j0qv1cbwh770zni3al8qxsphl353rv19f6rblplhs";
   };
 
-  nativeBuildInputs = [ makeWrapper menhir ];
-
-  propagatedBuildInputs = [ menhirLib merlin-extend ppx_tools_versioned ];
-
-  buildInputs = [ ocaml findlib dune_2 cppo fix utop menhir menhirSdk ];
+  nativeBuildInputs = [
+    makeWrapper
+    menhir
+  ];
+
+  propagatedBuildInputs = [
+    menhirLib
+    merlin-extend
+    ppxlib
+  ];
```

Comment:
Why is `ppxlib` propagated?

Reply:
Forgive my ignorance. I was replacing ocaml-migrate-parsetree which was propagated. I incorrectly thought this should be too. I'll fix it.

Reply:
Done.

Reply:
BTW, it seems that `ppx_derivers` and `result` should be propagated.

Reply:
I have created a demo (which likely should be added as a simple test) and it confirms the need of `ppx_derivers`. 
How do you know the need of `result`?

Reply:
Indeed, this is surprising. When removing `ppx_derivers` from the propagated inputs of `reason`, the error log reads:

>  161 |  (requires compiler-libs.common result ppx_derivers)
                                               ^^^^^^^^^^^^
  Error: Library "ppx_derivers" not found.
  -> required by library "reason.ocaml-migrate-parsetree" in

this suggests that `result` is also required…

Reply:
What if Result might be part of OCaml compiler already in newer version?

Reply:
Well, don’t add `result` if you believe it’s not necessary.

Reply:
Let's delay adding it until it breaks somewhere. I guess that will be the way to be sure of it.